presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1379,8 +1380,14 @@ private static HiveStorageFormat extractHiveStorageFormat(Table table)
 
     private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(Table table, TypeManager typeManager)
     {
+        Set<String> partitionKeyNames = table.getPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toSet());
+        List<FieldSchema> tableColumns = table.getSd().getCols();
+        if (tableColumns.stream().anyMatch(field -> partitionKeyNames.contains(field.getName()))) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Partition column cannot also be a table column");
```

Comment:
The question I have is what's the harm in having partition columns appear in the list of table columns (coming from the metastore). Is that indicative of corrupt metadata or is it an expected state that normally doesn't happen? Would it be safe to just ignore them? What does Hive do in this case?

So, the two options I see are:
* Change the code to be able to handle this scenario
* Fail with a clearer error message. Something like: Hive metadata for table xxx is corrupt. Partition keys are also listed in the storage descriptor"